signer-service/src/api/services/sep10.service.js

```diff
@@ -0,0 +1,76 @@
+const { Keypair } = require('stellar-sdk');
+const { TransactionBuilder, Networks } = require('stellar-sdk');
+const { fetchTomlValues } = require('../helpers/anchors');
+
+const { TOKEN_CONFIG } = require('../../constants/tokenConfig');
+const { SEP10_MASTER_SECRET } = require('../../constants/constants');
+
+const NETWORK_PASSPHRASE = Networks.PUBLIC;
+
+exports.signSep10Challenge = async (challengeXDR) => {
+  const masterStellarKeypair = Keypair.fromSecret(SEP10_MASTER_SECRET);
+  const { signingKey: anchorSigningKey } = await fetchTomlValues(TOKEN_CONFIG.ars.tomlFileUrl);
+
+  const transactionSigned = new TransactionBuilder.fromXDR(challengeXDR, NETWORK_PASSPHRASE);
+  if (transactionSigned.source !== anchorSigningKey) {
+    throw new Error(`Invalid source account: ${transactionSigned.source}`);
+  }
+  if (transactionSigned.sequence !== '0') {
+    throw new Error(`Invalid sequence number: ${transactionSigned.sequence}`);
+  }
+
+  // Verify manage_data operations
+  const operations = transactionSigned.operations;
+  // Verify the first manage_data operation
+  const firstOp = operations[0];
+  if (firstOp.type !== 'manageData') {
+    throw new Error('The first operation should be manageData');
+  }
+  // We only want to accept a challenge that would authorize the master key.
+  if (firstOp.source !== masterStellarKeypair.publicKey()) {
+    throw new Error('First manageData operation must have the master signing key as the source');
+  }
+
+  // Purposely hardcode Anclap's key, we don't want to sign this type of transaction
+  // for Mykobo
+  const expectedKey = `api.anclap.com auth`;
+  if (firstOp.name !== expectedKey) {
+    throw new Error(`First manageData operation should have key '${expectedKey}'`);
+  }
+  if (!firstOp.value || firstOp.value.length !== 64) {
+    throw new Error('First manageData operation should have a 64-byte random nonce as value');
+  }
+
+  // Flags to check presence of required operations
+  let hasWebAuthDomain = false;
+
+  // Verify extra manage_data operations
+  for (let i = 1; i < operations.length; i++) {
```

Comment:
Why not use operations.map() instead of a for loop?

Reply:
I found the explicit syntax more clear for this case such that it's visible what are we filtering for, but we could add a generic `verifyOperation` with all the rules and map over that, WDYT?
And return the first error it encounters.

Reply:
I think writing a function with a clear name and adding it to .map() can make the code more readable and self-documenting. Using a function `verifyOperation` with `.map` makes it easier to follow what is happening. wdyt?

Reply:
The issue is that it will require quite a lot of context and although it is easier to read, the control flow becomes harder as well as the errors returns. But let's see, I'll try to see how it looks.